sklearn/metrics/tests/test_common.py

```diff
@@ -1854,6 +1855,8 @@ def check_array_api_regression_metric(metric, array_namespace, device, dtype_nam
         metric_kwargs["sample_weight"] = np.array(
             [0.1, 2.0, 1.5, 0.5], dtype=dtype_name
         )
+    if func_name == "mean_tweedie_deviance":
+        metric_kwargs["power"] = -0.5
```

Comment:
Instead of hard-coding metric-specific parameters in the generic `check_array_api_regression_metric` checker, one should rather call `check_array_api_regression_metric` on `mean_tweedie_deviance` with various values for `power`, for instance using `functools.partial`.

```python
    mean_tweedie_deviance: [check_array_api_regression_metric],
    partial(mean_tweedie_deviance, power=-0.5): [check_array_api_regression_metric],
    partial(mean_tweedie_deviance, power=0): [check_array_api_regression_metric],
    ...
```

Reply:
Is this line `partial(mean_tweedie_deviance, power=0): [check_array_api_regression_metric],` necessary or is that already included with `mean_tweedie_deviance: [check_array_api_regression_metric],` (power is defaulted to 0 in the function definition)

Reply:
Ok to skip 0 but we need to make sure that all branches are tested for various choices of `power`.